src/screens/Voucher/VoucherDetailScreen.tsx

```diff
@@ -0,0 +1,90 @@
+import { StackScreenProps } from '@react-navigation/stack';
+import React, { useCallback, useContext, useState } from 'react';
+import { useQuery } from 'react-apollo';
+import { RefreshControl, ScrollView } from 'react-native';
+
+import { NetworkContext } from '../../NetworkProvider';
+import {
+  BoldText,
+  Discount,
+  HtmlView,
+  LoadingSpinner,
+  VoucherRedeem,
+  Wrapper
+} from '../../components';
+import { colors, texts } from '../../config';
+import { graphqlFetchPolicy } from '../../helpers';
+import { QUERY_TYPES, getQuery } from '../../queries';
+import { TVoucherContentBlock } from '../../types';
+
+export const VoucherDetailScreen = ({ route }: StackScreenProps<any>) => {
+  const { isConnected, isMainserverUp } = useContext(NetworkContext);
+  const fetchPolicy = graphqlFetchPolicy({ isConnected, isMainserverUp });
+  const [refreshing, setRefreshing] = useState(false);
+
+  const query = route.params?.query ?? '';
+  const queryVariables = route.params?.queryVariables ?? {};
+
+  const { data, loading, refetch } = useQuery(getQuery(query), {
+    variables: queryVariables,
+    fetchPolicy
+  });
+
+  const refresh = useCallback(async () => {
+    setRefreshing(true);
+    if (isConnected) {
+      await refetch();
+    }
+    setRefreshing(false);
+  }, [isConnected, refetch, setRefreshing]);
+
+  if (!data || loading) {
+    return <LoadingSpinner loading />;
+  }
+
+  const { contentBlocks, discountType, quota } = data[query];
+  const { availableQuantity, frequency, maxPerPerson, maxQuantity } = quota;
+
+  return (
+    <ScrollView
+      refreshControl={
+        <RefreshControl
+          refreshing={refreshing}
+          onRefresh={refresh}
+          colors={[colors.accent]}
+          tintColor={colors.accent}
+        />
+      }
+    >
+      <Wrapper>
+        {!!quota &&
+          (maxQuantity ? (
+            <BoldText small primary>
+              Limitiert: {availableQuantity}/{maxQuantity} verfügbar
+            </BoldText>
+          ) : (
+            <BoldText small primary>
+              {maxPerPerson}x pro Person {texts.voucher.detailScreen[frequency]} einlösbar
+            </BoldText>
+          ))}
```

Comment:
it could also be a combination of both

![image](https://github.com/smart-village-solutions/smart-village-app-app/assets/1942953/6d359f24-8330-47ae-9df3-b7fd60ae6a52)

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/989/commits/210403967ad95ec8aa4884dd59d387d34aa78168